fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -328,6 +373,24 @@ public Void visitArrayExpr(ArrayExpr node, Scope scope) {
             return null;
         }
 
+        @Override
+        public Void visitMapExpr(MapExpr node, Scope scope) {
+            if (!node.getChildren().isEmpty()) {
+                Type keyType = Type.NULL;
+                Type valueType = Type.NULL;
+                if (node.getKeyExpr() != null) {
+                    keyType = node.getKeyExpr().getType();
+                }
+                if (node.getValueExpr() != null) {
+                    valueType = node.getValueExpr().getType();
+                }
+                node.setType(new MapType(keyType, valueType));
+            } else {
+                node.setType(new MapType(Type.NULL, Type.NULL));
```

Comment:
set Type.BOOLEAN ?

Reply:
@Seaven Better to use TYPE.NULL here, because type comparatible machenism is based on NULL_TYPE and we can hack NULL_TYPE in the `ScalarOperatorToExpr.java` finally